### ISSUE TITLE
Fetching git repo fails on windows

### ISSUE THREAD

**Issue by h-vetinari** [conda/conda-build#4266]
A perfectly harmless source definition
```
source:
  # github tarball does not contain submodules, cf. dear-github/dear-github#214
  # in this case, we don't mind, because tomlplusplus is packaged separately...
  - url: https://github.com/bobfang1992/pytomlpp/archive/refs/tags/v{{ version }}.tar.gz
    sha256: 2aab6d0fa71b40e35c41ab783634d64d53b46c7d79c74251eb85ace9f3b61b25
    patches:
      - patches/0001-redirect-tomlplusplus-include-to-global-library-path.patch
  # ... however, we do include the toml-test submodule; keep in sync with
  # https://github.com/bobfang1992/pytomlpp/tree/master/tests/<commit>
  - git_url: https://github.com/BurntSushi/toml-test
    git_rev: 280497fa5f12e43d7233aed0d74e07ca61ef176b
    folder: tests/toml-test
```

in https://github.com/conda-forge/staged-recipes/pull/14903 fails on windows:
```
  File "C:\Miniconda\lib\site-packages\conda_build\source.py", line 333, in git_source
    git_depth=git_depth, is_top_level=True, verbose=verbose)
  File "C:\Miniconda\lib\site-packages\conda_build\source.py", line 251, in git_mirror_checkout_recursive
    assert isdir(mirror_dir)
AssertionError
```

I've read through [`source.py`](https://github.com/conda/conda-build/blob/master/conda_build/source.py), but I have no idea how the following can ever enter the else-clause, where the above error comes from:
https://github.com/conda/conda-build/blob/d2048eab3b095cca21d18fcae134e2d9b41ed2e6/conda_build/source.py#L208-L211

Shouldn't the `mirror_dir` always exist after lines 209-210? No idea where this is going wrong, but in combination with #4229, I'm blocked.

**Comment by cofinoa** [comment]
Hi @h-vetinari do you have any news on this issue?

**Comment by h-vetinari** [comment]
I don't have news unfortunately...
@mbargull @chenghlee could you support here, please?

**Comment by h-vetinari** [comment]
> could you support here, please?

Actually, let me rephrase that - if you have any idea about this, I'd appreciate if you could chime in. I don't mind raising a PR but I don't even understand what's going wrong.

**Comment by h-vetinari** [comment]
So things went back to working on windows (e.g. [here](https://github.com/conda-forge/pytomlpp-feedstock/blob/master/recipe/meta.yaml), and so I'm closing this.
